From 7a7b1bc46e72cd8c4cd0ca3123919405d20fe2b4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 28 Jan 2026 14:30:27 +0100
Subject: [PATCH] Fix compatibility with Python 3.14

Python 3.14 changing multiprocessing's start method on Linux from fork to
forkserver which requires parameters to be pickable.

Closes: https://gitlab.gnome.org/World/apostrophe/-/issues/659
---
 apostrophe.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/apostrophe.in b/apostrophe.in
index b71bcb1..5c5f683 100755
--- a/apostrophe.in
+++ b/apostrophe.in
@@ -19,6 +19,7 @@
 
 import sys
 import os
+import multiprocessing
 
 import gettext
 import locale
@@ -45,6 +46,7 @@ locale.bindtextdomain('apostrophe', localedir)
 gettext.textdomain('apostrophe')
 gettext.bindtextdomain('apostrophe', localedir)
 
+multiprocessing.set_start_method('fork')
 
 def run_application():
     from apostrophe.application import Application
-- 
2.52.0

